modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -139,165 +133,155 @@ PinMeshGenerator::PinMeshGenerator(const InputParameters & parameters)
-        params.set<std::vector<subdomain_id_type>>("ring_block_ids") = ring_blk_ids;
-        params.set<std::vector<unsigned int>>("ring_intervals") = ring_intervals;
+        background_blk_ids.insert(background_blk_ids.begin(), background_blk_ids.front());
+        background_blk_names.insert(background_blk_names.begin(), background_blk_names.front());
       }
-
-      params.set<std::vector<subdomain_id_type>>("background_block_ids") = background_ids;
-      params.set<unsigned int>("background_intervals") = background_intervals;
-
-      if (duct_intervals.size() > 0)
+      else
       {
-        params.set<MooseEnum>("duct_sizes_style") = "apothem";
-        params.set<std::vector<Real>>("duct_sizes") = _duct_halfpitch;
-        params.set<std::vector<subdomain_id_type>>("duct_block_ids") = duct_ids;
-        params.set<std::vector<unsigned int>>("duct_intervals") = duct_intervals;
+        const auto region_id = _region_ids[0][0];
+        const auto block_name = background_blk_names.front() + "_TRI";
+        const auto block_id = getBlockId(block_name, region_id);
+        background_blk_ids.insert(background_blk_ids.begin(), block_id);
+        background_blk_names.insert(background_blk_names.begin(), block_name);
       }
-      params.set<MooseEnum>("polygon_size_style") = "apothem";
-      params.set<Real>("polygon_size") = _pitch / 2.0;
+    }
+    // Add _TRI suffix if only one background region and tri center elements
+    // and no ring regions
+    else if (!_quad_center)
+      background_blk_names[0] += "_TRI";
+  }
 
-      _build_mesh =
-          &addMeshSubgenerator("PolygonConcentricCircleMeshGenerator", name() + "_2D", params);
+  // Generate Cartesian/hex pin using PolygonConcentricCircleMeshGenerator
+  {
+    // Get and assign parameters for the main geometry feature of the Pin
+    // which is created with a PolygonConcentricCircleMeshGenerator subgenerator
+    auto params = _app.getFactory().getValidParams("PolygonConcentricCircleMeshGenerator");
+    params.set<bool>("preserve_volumes") = true;
+    params.set<bool>("quad_center_elements") = _quad_center;
+    params.set<MooseEnum>("polygon_size_style") = "apothem";
+    params.set<Real>("polygon_size") = _pitch / 2.0;
+    params.set<boundary_id_type>("external_boundary_id") = 20000 + _pin_type;
+    params.set<std::string>("external_boundary_name") = "outer_pin_" + std::to_string(_pin_type);
+    bool flat_side_up = (_mesh_geometry == "Square");
+    params.set<bool>("flat_side_up") = flat_side_up;
+
+    const auto num_sides = (_mesh_geometry == "Square") ? 4 : 6;
+    params.set<unsigned int>("num_sides") = num_sides;
+    params.set<std::vector<unsigned int>>("num_sectors_per_side") =
+        std::vector<unsigned int>(num_sides, _num_sectors);
+
+    params.set<boundary_id_type>("interface_boundary_id_shift") =
+        30000; // need to shift interface boundaries to avoid clashing
+               // with default IDs for PatternedMeshGenerator
```

Comment:
is this relevant for hex pins?

Reply:
Turns out this isn't needed for Cartesian or hex pins so I've gone ahead and removed this block